db/compaction_job.cc

```diff
@@ -282,6 +282,7 @@ CompactionJob::CompactionJob(
       db_options_(db_options),
       env_options_(env_options),
       env_(db_options.env),
+      env_options_for_compaction_(env_->OptimizeForCompactionTableRead(env_options, db_options_)),
```

Comment:
I believe one of `env_options_` and `env_options_for_compaction_` is for read and the other is for write. Can you rename them `env_optiosn_for_read_` and `env_options_for_write_` and double check all their usage in compaction job is correct?